src/authorization.ts

```diff
@@ -31,8 +36,5 @@ export interface Authorization {
     /**
      * Returns the resourceTypes for which the requester is allowed to perform the given operation.
      */
-    resourceTypesAllowedForOperation(
-        accessToken: string,
-        operation: TypeOperation | SystemOperation,
-    ): Promise<string[]>;
+    getAllowedResourceTypesForOperation(request: getAllowedResourceTypesForOperationRequest): string[];

```

Comment:
One thing we haven't really talked about is when should we return `Promise<type>` or not? thoughts? My feeling is default to `Promise` unless we have a reason not to

Reply:
I agree that `Promise<T>` is better for the interface. That keeps the interface open for implementations that may want to do some async calls (i.e. to an auth server)

The only thing is that the most important method (`isAuthorized`) is synchronous right now. I actually made this method async first but then switched to sync for consistency since it can be derived from `isAuthorized`.

Reply:
Since this is a breaking change anyways I say lets make both methods async

Reply:
I'll do that.

Turns out that `isAuthorized` is easy to refactor across the other packages